exporter/trace/otelcol/connection.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelcol
+
+import (
+	"math/rand"
+	"sync/atomic"
+	"time"
+	"unsafe"
+)
+
+func (e *Exporter) lastConnectError() error {
+	errPtr := (*error)(atomic.LoadPointer(&e.lastConnectErrPtr))
+	if errPtr == nil {
+		return nil
+	}
+	return *errPtr
+}
+
+func (e *Exporter) saveLastConnectError(err error) {
+	var errPtr *error
+	if err != nil {
+		errPtr = &err
+	}
+	atomic.StorePointer(&e.lastConnectErrPtr, unsafe.Pointer(errPtr))
+}
+
+func (e *Exporter) setStateDisconnected(err error) {
+	e.saveLastConnectError(err)
+	select {
+	case e.disconnectedCh <- true:
+	default:
+	}
+}
+
+func (e *Exporter) setStateConnected() {
+	e.saveLastConnectError(nil)
+}
+
+func (e *Exporter) connected() bool {
+	return e.lastConnectError() == nil
+}
+
+const defaultConnReattemptPeriod = 10 * time.Second
+
+func (e *Exporter) indefiniteBackgroundConnection() error {
+	defer func() {
+		e.backgroundConnectionDoneCh <- true
+	}()
+
+	connReattemptPeriod := e.reconnectionPeriod
+	if connReattemptPeriod <= 0 {
+		connReattemptPeriod = defaultConnReattemptPeriod
+	}
+
+	// No strong seeding required, nano time can
+	// already help with pseudo uniqueness.
+	rng := rand.New(rand.NewSource(time.Now().UnixNano() + rand.Int63n(1024)))
+
+	// maxJitter: 1 + (70% of the connectionReattemptPeriod)
+	maxJitter := int64(1 + 0.7*float64(connReattemptPeriod))
```

Comment:
This is in nanoseconds, as far as I understand. Maybe rename to `maxJitterNanos` or similar to make it clear?
Also, what's the role of extra nanosecond here?

Reply:
done. remove extra nanosecond.